src/properties/enum_property.rs

```diff
@@ -34,10 +34,7 @@ impl EnumProperty {
         if include_header {
             Self::read_header(cursor)
         } else {
-            Err(DeserializeError::invalid_property(
-                "EnumProperty is not supported in arrays",
-                cursor,
-            ))?
+            Self::read_body(cursor, String::from("<array>"))
```

Comment:
I had some issues using an Optional for `enum_type`. It might be possible to refactor it to an `Optional<&str>`, which would work with the `impl_write!` macro and `write_fstring`. More effort is required to get this working.